### PR TITLE
Fixed pokemon showing up double in getCatchablePokemon()

### PR DESCRIPTION
**Pull request by Jari27**
This should fix it.

### CODE REVIEW

**Comment by przerwap** [comment]
Would it not be simpler to just collect the pokemon into a Set

**Comment by Jari27** [comment]
Probably not, since then you'd still have to check against their encounterId (`getEncounterId()`), since the actual objects aren't the same. Instead of breaking working things I decided to do it this way.

**Comment by przerwap** [comment]
Equality of objects is determined though the equals() method. And Sets can't contain duplicates.

**Comment by Jari27** [comment]
So you are suggesting changing the list to a collection, changing every piece of code that depends on that list being a list and overriding the `equals`method in the `CatchablePokemon`class?

Obviously that's a solution, but like I said before: I'm not breaking things that are working if I can help it.

**Comment by przerwap** [comment]
Nah man. There are basically two thing you can do. Each includes defining that simple equals() and hashCode() method.

You can Just make a set .. which makes it so you don't have to do all that logic there like so:

``` Java
Set<CatchablePokemon> pokemon = new HashSet<>(catchablePokemons);

return new ArrayList<>(pokemon);
```

Or .. if you look a few lines down in the same class, you'll notice that there are Streams being used in the project.

``` Java
return RefStreams.concat(
        StreamSupport.stream(objects.getCatchablePokemons())
                .map(new Function<MapPokemon, CatchablePokemon>() {
                    @Override
                    public CatchablePokemon apply(MapPokemon mapPokemon) {
                        return new CatchablePokemon(api, mapPokemon);
                    }
                }),
        StreamSupport.stream(objects.getWildPokemons())
                .map(new Function<WildPokemonOuterClass.WildPokemon, CatchablePokemon>() {
                    @Override
                    public CatchablePokemon apply(WildPokemonOuterClass.WildPokemon mapPokemon) {
                        return new CatchablePokemon(api, mapPokemon);
                    }
                }))
        .distinct()
        .collect(Collectors.<CatchablePokemon>toList());
```

The second option is likely preferable I'd say, since it removes the need for those for loops mapping the Wild/Map Pokemon classes to CatchablePokemon. It's a shame we can't use actual lambda here, but oh well.

**Comment by jabbink** [comment]
I think replacing `Collection<CatchablePokemon>` by `Set<CatchablePokemon>` and implementing the hashCode and equals method is quite elegant in this case.

**Comment by Grover-c13** [comment]
I agree a set would be best, hashCode can just use the encounterid since it will be unique.

**Comment by Jari27** [comment]
Done.

**Comment by FabianTerhorst** [comment]
you have to rebase from the masterbranch und fix the codestyle inside the catchablepokemon.java

**Comment by FabianTerhorst** [comment]
whats about: 

```
 if (o == this) {
    return true;
 } else if (o instanceof CatchablePokemon) {
        return getEncounterId() == ((CatchablePokemon) o).getEncounterId();
 }
return false
```

for https://github.com/Grover-c13/PokeGOAPI-Java/pull/134/files#diff-2abf18e8affc55dadaab37add72c02adR223
?

File: src/main/java/com/pokegoapi/api/map/Map.java [hunk 1]
```diff
@@ -91,8 +95,8 @@ public void clearCache() {
 	 *
 	 * @return a List of CatchablePokemon at your current location
 	 */
-	public List<CatchablePokemon> getCatchablePokemon() throws LoginFailedException, RemoteServerException {
-		List<CatchablePokemon> catchablePokemons = new ArrayList<>();
+	public Set<CatchablePokemon> getCatchablePokemon() throws LoginFailedException, RemoteServerException {
```
Comment (przerwap) [inline src/main/java/com/pokegoapi/api/map/Map.java]: You can't change the response type here. That would affect anyone/anything that is currently using this method.

Comment (FabianTerhorst) [inline src/main/java/com/pokegoapi/api/map/Map.java]: He could return a list and transform the set into a arraylist with

```
new ArrayList<CatchablePokemon>(catchablePokemons);
```

**Comment by Grover-c13** [comment]
Please rebase & conform to the style check (gradlbe build should report issues)

**Comment by Jari27** [comment]
Uuuuhg it's messed up.

File: build.gradle [hunk 2]
```diff
@@ -16,7 +16,6 @@ apply plugin: 'idea'
 apply plugin: 'java'
 apply plugin: 'maven'
 apply plugin: 'com.google.protobuf'
-apply plugin: 'checkstyle'
```
Comment (jabbink) [inline build.gradle:19]: That's a way to fix the checkstyle.

Comment (Jari27) [inline build.gradle:19]: I can't get checkstyle to work with my java version so I turned it off, but now it's committing the build.gradle file as well. Haven't worked with git that often :(

Comment (jabbink) [inline build.gradle:19]: `git commit -a` sucks

Comment (Jari27) [inline build.gradle:19]: Didn't even use that. I think it came from the rebasing. I only changed (and added and committed) Map.java and CatchablePokemon.java.

Comment (Jari27) [inline build.gradle:19]: And it seems there are errors in the code I pulled, which is weird.

**Comment by Jari27** [comment]
I'm starting over.

**Pull request closed without merging**